product_tier_validation/data/product_state_data.xml

```diff
@@ -1,8 +1,30 @@
 <odoo noupdate="1">
+    <record id="product_state_cancel" model="product.state">
+        <field name="code">cancel</field>
+        <field name="name">Cancelled</field>
+        <field name="sequence">50</field>
+    </record>
+
+    <function name="write" model="ir.model.data">
+        <function name="search" model="ir.model.data">
+            <value eval="[('module', '=', 'product_state')]" />
+        </function>
+        <value eval="{'noupdate': False}" />
+    </function>
```

Comment:
instead of 3 writes can't you just write default=False on the matching record?
Not a blocker ;)

Reply:
Thanks for the great advice! The PR has been updated.